02.calendar/calendar.rb

```diff
@@ -10,13 +10,14 @@
 
 puts "日 月 火 水 木 金 土"
 
-first_wday = Date.new(year, month, 1).wday
-last_day = Date.new(year, month, -1).day
-print "   " * first_wday
+first_date = Date.new(year, month, 1).day
+last_date = Date.new(year, month, -1).day
+first_day = Date.new(year, month, 1).wday
+print "   " * first_day
 
-(1..last_day).each { |day|
-  printf("%2d ", day)
-    if (first_wday + day) % 7 == 0 || day == last_day
+(first_date..last_date).each do |date|
+  printf("%2d ", date)
+  if (first_day + date) % 7 == 0
```

Comment:
Dateオブジェクトを活用すれば、この行もこんなふうに書けるはずです。

```suggestion
  if date.saturday?
```